dashboard/app/models/course_version.rb

```diff
@@ -18,4 +18,31 @@
 
 class CourseVersion < ApplicationRecord
   belongs_to :content_root, polymorphic: true
+
+  def self.update_course_version(content_root)
+    # TODO: Once the Course model is added, change this to just be version_year, since then the
```

Comment:
Does this seem right? I was thinking having the family name in the key will be redundant long-term.

Reply:
I'm wondering why CourseVersion would not be the owner of version year?

Reply:
> I'm wondering why CourseVersion would not be the owner of version year instead of script

IIRC the way we shortened the timeline for this undertaking was to have these properties continue to live on the content root, and as we add new features, we could look for them on the CourseVersion and the CourseVersion could delegate the call to the content root. it would look like this:
```
class CourseVersion < ApplicationRecord
  delegate :family_name, to: :content_root
```
And then instance methods (but not this class method), could just call `family_name` instead of `content_root.family_name`. 

In the future, we would move the family_name fully into the CourseVersion, in which case I'm not sure how the code below would need to be updated because the content_root here might not have a Course Version yet.

Reply:
> `# TODO: Once the Course model is added, change this to just be version_year, since then the`
>  `# unique index will be on (course_id, key).`

> Does this seem right? I was thinking having the family name in the key will be redundant long-term.

This makes sense to me. Once we have created the new Course model, won't it make more sense to have version_year be it's own db field and just index directly on (course_id, version_year), though?

Reply:
In that option, would we just get rid of `key`? I was thinking that logically it'd be the same as what you're saying, we'd just call it `key`.

Reply:
agreed, I think those two solutions would be equivalent, just with the column named either `version_year` or `key`.